twilight-model/src/application/command/option.rs

```diff
@@ -182,8 +170,30 @@ pub struct CommandOptionChoiceData<T> {
     /// [`name`]: Self::name
     #[serde(skip_serializing_if = "Option::is_none")]
     pub name_localizations: Option<HashMap<String, String>>,
-    /// Value of the choice. Must be 100 characters or less if it is a string.
-    pub value: T,
+
+    /// Value of the choice.
+    pub value: CommandOptionChoiceValue,
+}
+
+/// The value of a [`CommandOptionChoice`].
+///
+/// Note that the right variant must be selected based on the
+/// [`CommandOption`]'s [`CommandOptionType`].
+///
+/// See [`CommandOptionChoice`]'s documentation for more info.
```

Comment:
There's not really any more info there (also remove the intralink at the bottom)
```suggestion
```